static/widgets/history-widget.ts

```diff
@@ -26,80 +26,41 @@ import $ from 'jquery';
 import {pluck} from 'underscore';
 import {sortedList, HistoryEntry, EditorSource} from '../history.js';
 import {editor} from 'monaco-editor';
-
-import IStandaloneDiffEditor = editor.IStandaloneDiffEditor;
 import ITextModel = editor.ITextModel;
 import {unwrap} from '../assert.js';
 
-export class HistoryDiffState {
-    public model: ITextModel;
-    private result: EditorSource[];
-
-    constructor(model: ITextModel) {
-        this.model = model;
-        this.result = [];
-    }
-
-    update(result: HistoryEntry) {
-        this.result = result.sources;
-        this.refresh();
-
-        return true;
-    }
-
-    private refresh() {
-        const output = this.result;
-        const content = output.map(val => `/****** ${val.lang} ******/\n${val.source}`).join('\n');
-
-        this.model.setValue(content);
-    }
-}
-
 type Entry = {dt: number; name: string; load: () => void};
 
 export class HistoryWidget {
     private modal: JQuery | null;
-    private diffEditor: IStandaloneDiffEditor | null;
-    private lhs: HistoryDiffState | null;
-    private rhs: HistoryDiffState | null;
+    private srcDisplay: editor.ICodeEditor | null;
+    private model: ITextModel;
     private currentList: HistoryEntry[];
     private onLoadCallback: (data: HistoryEntry) => void;
 
     constructor() {
         this.modal = null;
-        this.diffEditor = null;
-        this.lhs = null;
-        this.rhs = null;
+        this.srcDisplay = null;
```

Comment:
Maybe just `private srcDisplay: editor.ICodeEditor | null = null;` above?

Reply:
Thanks! I now did this for the initialization of all 5 private data members.  It might be slightly 'foreign' to the rest of CE but I like it better too.